File: src/etch/examples/ccmuser_cs/server/ServerMain.cs
using System;
using System.Collections.Generic;
using System.Text;

using Etch.Transport;
using Etch.Msg;

using etch.examples.ccmuser;

namespace etch.examples.ccmuser.server
{
    class ServerMain
    {
        public ListenerHandler lh;

        public ServerMain()
        {
            lh = new MyDefaultListenerHandler(this);
        }

        public class MyDefaultListenerHandler : DefaultListenerHandler
        {
            ServerMain sm;
            MyDeliveryService sds;

            public MyDefaultListenerHandler(ServerMain sm)
            {
                this.sm = sm;
                this.sds = new MyDeliveryService(this);
            }

            protected override Etch.Transport.MessageHandler<MailboxMessageSource> NewMessageHandler()
            {
                return new MyDeliveryService(this);
            }

            protected override ValueFactory NewValueFactory()
            {
                return new ValueFactoryCcmUser();
            }

            public override string ToString()
            {
                return "MyDefaultListenerHandler";
            }
            
            public class MyDeliveryService : DefaultDeliveryService<StubCcmUserServer>
            {
                MyDefaultListenerHandler sdlh;

                public MyDeliveryService(MyDefaultListenerHandler sdlh)
                {
                    this.sdlh = sdlh;
                }

                protected override StubCcmUserServer NewStub(MailboxMessageSource src)
                {
                    this.SetMailboxMessageSource(src);
                    RemoteCcmUserClient rsc = new RemoteCcmUserClient(this);
                    return new StubCcmUserServer(new CcmUserServerImpl(rsc), null, null);
                }

                public override string ToString()
                {
                    return "MyDeliveryService";
                }
            }

        }

        public static void Main(String[] args)
        {
            ServerMain sm = new ServerMain();
            Listener l = new Listener(sm.lh, 5, "127.0.0.1", 4021, 0);
            l.Start();
            l.WaitUp(4000);
        }
    }
}

File: src/etch/examples/benchmark/simple/src/etch/examples/simple/ServerMain.java
package etch.examples.simple;

import metreos.core.io.Listener;
import metreos.core.io.ListenerHandler;
import etch.bindings.java.msg.ValueFactory;
import etch.bindings.java.transport.DefaultDeliveryService;
import etch.bindings.java.transport.DefaultListenerHandler;
import etch.bindings.java.transport.DefaultMessageHandler;
import etch.bindings.java.transport.MailboxMessageSource;
import etch.bindings.java.transport.MessageHandler;

public class ServerMain 
{
    public static void main( String[] args ) throws Exception
    {
        ListenerHandler lh = new DefaultListenerHandler()
        {
            @Override
            protected MessageHandler<MailboxMessageSource> newMessageHandler()
            {
                return new DefaultMessageHandler<StubSimpleServer>()
                {
                    @Override
                    protected StubSimpleServer newStub( MailboxMessageSource src )
                    {
                        DefaultDeliveryService svc = new DefaultDeliveryService(src, null);
                        RemoteSimpleClient rfc = new RemoteSimpleClient(svc);
                        return new StubSimpleServer(new SimpleServerImpl(rfc), null, null);
                    }

                    @Override
                    public String toString()
                    {
                        return "MessageHandler";
                    }
                };
            }

            @Override
            protected ValueFactory newValueFactory()
            {
                return new ValueFactorySimple();
            }

            @Override
            public String toString()
            {
                return "ListenerHandler";
            }
        };
        
        Listener l = new Listener( lh, 5, "", 4001, 0 );
        l.start();
    }
}